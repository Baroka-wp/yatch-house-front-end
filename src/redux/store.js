import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import houseReducer, { getAllItems } from './house/houseReducer';
import reservationReducer from './reservation/reservationReducer';

const rootReducer = combineReducers({
  houses: houseReducer,
  reservations: reservationReducer,
});
const store = createStore(rootReducer, applyMiddleware(thunk));
store.dispatch(getAllItems());

export default store;
