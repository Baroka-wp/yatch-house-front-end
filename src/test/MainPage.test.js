import { Provider } from 'react-redux';
// import MainPage from '../components/pages/mainpage';
import store from '../redux/store';

describe('mainpage', () => {
  it('renders mainpage component', () => {
    const data = [{
      id: 1,
      name: 'House 1',
      description: 'House 1 description',
      price: 100,
      image: 'https://picsum.photos/200/300',
    }];

    const MainPage = () => (
      <Provider store={store}>
        <div>
          <h1>{data[0].name}</h1>
          <p>{data[0].description}</p>
        </div>
      </Provider>
    );

    expect(MainPage).toMatchSnapshot();
  });
});
