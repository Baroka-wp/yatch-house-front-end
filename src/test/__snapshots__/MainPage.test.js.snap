// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainpage renders mainpage component 1`] = `[Function]`;
